docs/demos/UAV_tutorial.py

```diff
@@ -0,0 +1,279 @@
+
+# %%
+# Setup Deletor for the Tracker
+# -----------------------------
+# In the simple case of 1 target, we never want to delete the track. Because
+# this Deletor is so simple I haven't bothered using a subtype/inheritance.
+# Instead I make use of Python's duck typing. 
+class MyDeleter:
+    def delete_tracks(self, tracks):
+        return set()
+ 
+deleter = MyDeleter()
+
+# %%
+# Setup Hypothesiser and Associator
+# ---------------------------------
+# Since we know there is only one measurement per scan, we can just use the
+# :class:`~.NearestNeighbour` associator to achieve our desired result.
+from stonesoup.measures import Euclidean
+from stonesoup.dataassociator.neighbour import NearestNeighbour
+from stonesoup.hypothesiser.distance import DistanceHypothesiser
+from stonesoup.tracker.simple import SingleTargetTracker
+meas = Euclidean()
+hypothesiser = DistanceHypothesiser(predictor, updater, meas)
+associator = NearestNeighbour(hypothesiser)
+
+
+tracker = SingleTargetTracker(initiator,
+                              deleter,
+                              detector,
+                              associator,
+                              updater)
+
+# %%
+# Run the Tracker
+# ---------------------------------
+# We extract the ground truth from the detector and then run the tracker.
+# While running the tracker we:
+#
+# - Extract the measurement that is associated with it.
+# - Extract the position components of the estimated state vector.
+#
+# This allows us to plot the measurements, ground truth, and state estimates.
+#
+# **Note:** The meas_model.inverse_function() returns a state vector, which
+# for our CV model consists of [x, vx, y, vy, z, vz].
+from matplotlib import pyplot as plt
+est_X=[]
+est_Y=[]
+meas_X=[]
+meas_Y=[]
+true_X = []
+true_Y = []
+for time, tracks in tracker:
+    for ground_truth in ground_truth_reader.groundtruth_paths:
+        true_X.append(ground_truth.state_vector[0])
+        true_Y.append(ground_truth.state_vector[2])
+
+    # Because this is a single target tracker, I know there is only 1 track.
+    for track in tracks:
+
+        #Get the corresponding measurement
+        detection = track.states[-1].hypothesis.measurement
+        # Convert measurement into xy
+        xyz = meas_model.inverse_function(detection)
+        meas_X.append(xyz[0])
+        meas_Y.append(xyz[2])
+        
+        vec = track.states[-1].state_vector
+        est_X.append(vec[0])
+        est_Y.append(vec[2])
+
+fig = plt.figure(figsize=(10, 6))
+ax1 = fig.add_subplot(1, 1, 1)
+plt.plot(meas_X, meas_Y, 'xb', label='Measurements')
+ax1.plot(true_X, true_Y, 'd-k', label='Truth', markerfacecolor='None')
+ax1.legend()
+ax1.set_xlabel('X (m)')
+ax1.set_ylabel('Y (m)')
+
+fig = plt.figure(figsize=(10, 6))
+ax2 = fig.add_subplot(1, 1, 1)
+ax2.plot(true_X, true_Y, 'd-k', label='Truth', markerfacecolor='None')
+ax2.plot(est_X, est_Y, 'r.', label='Estimates')
+ax2.set_xlabel('X (m)')
+ax2.set_ylabel('Y (m)')
+ax2.legend()
```

Comment:
I tried overlaying the measurements and the estimates, but the graph is too busy. The measurements are very noisy and because the track comes back on itself, you're likely to infer the wrong measurement with the estimate. This works better when you only show a few measurements at a time. Using an animated sequence would be better, but that also brings other difficulties with it. Similar problem with the uncertainties - it's just too busy with so many measurements.